os/entropystat/entropystat_linux.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015 Square, Inc
+
+// Package entropystat implements metrics collection for system entropy
+package entropystat
+
+import (
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/square/inspect/metrics"
+	"github.com/square/inspect/os/misc"
+)
+
+// to make testing easy
+var path = "/proc/sys/kernel/random/entropy_avail"
+
+// EntropyStat represents available entropy on the system
+type EntropyStat struct {
+	Available *metrics.Gauge
+	m         *metrics.MetricContext
+}
+
+// New starts metrics collection every Step and registers with
+// metricscontext
+func New(m *metrics.MetricContext, Step time.Duration) *EntropyStat {
+	stat := new(EntropyStat)
+	stat.m = m
+	// initialize all metrics and register them
+	misc.InitializeMetrics(stat, m, "entropystat", true)
+	// collect once
+	stat.Collect()
```

Comment:
Pretty sure this isn't necessary.

Reply:
All the other New() methods do this

Reply:
otherwise we'd wait for the step duration before we collect the first stat, and it wouldn't be in the returned stat object

Reply:
well not all, apparently: seems mixed (some do, some don't).

Reply:
Also, if you don't do this, it looks harder to test.